glyphs2fontir/src/source.rs

```diff
@@ -137,6 +144,76 @@ impl GlyphsIrSource {
     }
 }
 
+struct StaticMetadataWork {
+    font: Arc<Font>,
+}
+
+impl Work for StaticMetadataWork {
+    fn exec(&self, context: &Context) -> Result<(), WorkError> {
+        let font = self.font.as_ref();
+        debug!("Static metadata for {}", font.family_name);
+
+        let mut axis_values = Vec::new();
+        for master in font.font_master.iter() {
+            master
+                .axes_values
+                .as_ref()
+                .ok_or_else(|| WorkError::InconsistentAxisDefinitions)?
+                .iter()
+                .enumerate()
+                .for_each(|(idx, value)| {
+                    while axis_values.len() < idx {
+                        axis_values.push(Vec::new());
+                    }
+                    axis_values[idx].push(value);
+                });
+        }
+
+        let axes = font.axes.as_ref().ok_or(WorkError::NoAxisDefinitions)?;
+        if axes.is_empty() {
+            return Err(WorkError::NoAxisDefinitions);
+        }
+        if axis_values.iter().any(|av| axes.len() != av.len()) {
+            return Err(WorkError::InconsistentAxisDefinitions);
+        }
+
+        let default_master_idx = font.default_master_idx();
+
+        let axes = axes
+            .iter()
+            .enumerate()
+            .map(|(idx, a)| {
+                let min = axis_values[idx]
+                    .iter()
+                    .map(|v| OrderedFloat::<f32>(v.into_inner() as f32))
+                    .min()
+                    .unwrap();
+                let max = axis_values[idx]
+                    .iter()
+                    .map(|v| OrderedFloat::<f32>(v.into_inner() as f32))
+                    .max()
+                    .unwrap();
+                let default =
+                    OrderedFloat::<f32>(axis_values[default_master_idx][idx].into_inner() as f32);
+
+                Axis {
+                    name: a.name.clone(),
+                    tag: a.tag.clone(),
+                    hidden: a.hidden.unwrap_or(false),
+                    min,
```

Comment:
ok, so this immediately brings https://github.com/googlefonts/fontmake-rs/issues/22 to the front of the queue..
The master axis values in Glyphs are in internal design coordinates. The global axes definitions (the min/default/max that go in fvar, the .designspace axes min/default/max, and hence also our ir:Axis) are in user-space coordinates.
The mapping between user-space to internal design-space coordinates is defined in Glyphs in multiple ways

- directly via the "Axis Mappings" font custom parameter
- via "Axis Location" custom parameters set on each font master
- implicitly by deriving it for wght and wdth axis only from the user-space location of the instances

it's one of the most complicated pieces of code in glyphsLib, for your amusement 

https://github.com/googlefonts/glyphsLib/blob/f6e9c4a29ce764d34c309caef5118c48c156be36/Lib/glyphsLib/builder/axes.py#L155

Reply:
I copied ^ to #22 because I want to deal with it in subsequent PR(s)

Reply:
and *not* deal with it here, so resolving